packages/object-safe-hasownproperty/index.js

```diff
@@ -0,0 +1,62 @@
+module.exports = hop;
+
+/*
+  const obj = {a: {aa: {aaa: 2}}, b: 4};
+
+  hop(obj, 'a.aa.aaa'); // true
+  hop(obj, ['a', 'aa', 'aaa']); // true
+
+  hop(obj, 'b.bb.bbb'); // false
+  hop(obj, ['b', 'bb', 'bbb']); // false
+
+  hop(obj.a, 'aa.aaa'); // true
+  hop(obj.a, ['aa', 'aaa']); // true
+
+  hop(obj.b, 'bb.bbb'); // false
+  hop(obj.b, ['bb', 'bbb']); // false
+
+  hop(null, 'a'); // false
+  hop(undefined, ['a']); // false
+
+  const obj = {a: {}};
+  const sym = Symbol();
+  obj.a[sym] = 4;
+  hop(obj.a, sym); // true
+*/
+
+function hop(obj, propsArg) {
+  if (!obj) {
+    return false;
+  }
+  var props, prop;
+  if (Array.isArray(propsArg)) {
+    props = propsArg.slice(0);
+  }
+  if (typeof propsArg == 'string') {
+    props = propsArg.split('.');
+  }
+  if (typeof propsArg == 'symbol') {
+    props = [propsArg];
+  }
+  if (!Array.isArray(props)) {
+    throw new Error('props arg must be an array, a string or a symbol');
+  }
+
+  while (props.length) {
+    prop = props.shift();
+
+    // if we are recursing, but met a nullish value, we cannot
+    // access it via .hasOwnProperty and should return negatively
+    if (obj === null || obj === undefined) return false;
+
+    if (!Object.prototype.hasOwnProperty.call(obj, prop)) {
+      return false;
+    }
+
+    if (props.length === 0) return true;
```

Comment:
house style is {} and new line for `if` statements

Reply:
done. Also removed newlines between blocks (as the ones above where styled)